.changeset/khaki-glasses-raise.md

```diff
@@ -0,0 +1,48 @@
+---
+'astro': minor
+---
+
+## Integration Hooks to add Middleware
+
+It's now possible in Astro for an integration to add middleware on behalf of the user. Previously when a 3rd party wanted to provide middleware the user would need to create a `src/middleware.ts` file themselves. Now adding 3rd party middleware is as easy as adding a new integration.
```

Comment:
```suggestion
It's now possible in Astro for an integration to add middleware on behalf of the user. Previously when a third party wanted to provide middleware, the user would need to create a `src/middleware.ts` file themselves. Now, adding third-party middleware is as easy as adding a new integration.
```
As if you were trying to test me, Matthew.  😅 

To be extra clear that this is intentional, yes there is a hyphen required only in the second use of "third-party."